dracut/99setup-root/initrd-setup-root-after-ignition

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+set -euo pipefail
+
+# Manage sysext symlinks to select the active sysext image, and do OEM partition migrations
+# (An alternative to symlinks is to implement support for ".v directories" in systemd-sysext
+# that could contain multiple versions and the right one gets selected, here not the newest
+# but that matching the OS version. However, we would still need the migration logic and
+# would not be able to use the OEM partition as dynamically preferred storage.)
+VERSION=$(grep -m 1 "^VERSION=" /sysroot/usr/lib/os-release | cut -d = -f 2)
+
+# Not all OEM partitions have an oem-release file but we require it now
+OEMID=$({ grep -m 1 -o "^ID=.*" /sysroot/usr/share/oem/oem-release || true ; } | cut -d = -f 2)
+
+# The active-oem-OEMID file gets created by the update-engine postinst action if both old and new /usr partitions have a sysext
+if [ "${OEMID}" != "" ] && [ -e "/sysroot/usr/share/oem/sysext/active-oem-${OEMID}" ]; then
+  SYSEXT_OEM_PART="/usr/share/oem/sysext/oem-${OEMID}-${VERSION}.raw"
+  SYSEXT_ROOT_PART="/etc/flatcar/oem-sysext/oem-${OEMID}-${VERSION}.raw"
+  SYMLINK="/sysroot/etc/extensions/oem-${OEMID}.raw"
+  ACTIVE_OEM=""
+
+  if [ -e "/sysroot/${SYSEXT_OEM_PART}" ]; then
+    ACTIVE_OEM="${SYSEXT_OEM_PART}"
+  elif [ -e "/sysroot/${SYSEXT_ROOT_PART}" ]; then
+    if [ -L "${SYMLINK}" ]; then
+      # Move the inactive sysext to the root partition to free space on the OEM partition
+      PREV_SYSEXT=$(readlink "${SYMLINK}")
```

Comment:
Is `realpath` available in initrd? You could use it, so you'll certainly get an absolute path.

Reply:
Thanks. It's not there but with `--relative-to=/sysroot` (if I understand correctly) it would be nice to resolve the link properly. I'll test it by calling it with LD_LIBRARY_PATH from /sysroot/usr/bin/.

Reply:
There is `dracut_install grep ldconfig mountpoint systemd-tmpfiles flatcar-tmpfiles` in `module-setup.sh` in the same directory as this script. You could add `realpath` there too. It comes from coreutils, and we are already putting some coreutils stuff into initrd, so adding `realpath` shouldn't hurt.

Reply:
It doesn't really help much, with `-m --relative-base=/sysroot` I get the into a similar situation as with `readlink -m` where depending on whether it's an absolute link it gets a prefix or not…

Reply:
Fair enough.

Reply:
Updated it with a `sed` workaround